src/index.js

```diff
@@ -72,6 +84,22 @@ var Index = new Lang.Class({
             doc.add_numeric_value(SEQUENCE_NUMBER_VALUE_NO, metadata['sequenceNumber']);
         }
 
+        const indexingDateString = _indexDateForMetadata(metadata);
+        if (indexingDateString) {
+            /* Parse the date using the native JS Date library, since
+             * GLib.Date.set_parse isn't able to handle JS-style dates. */
+            try {
+                const parsedIndexingDate = new Date(indexingDateString);
```

Comment:
In order to avoid a confusing unused variable, you could do
```js
void new Date(indexingDateString);
```

Reply:
Or did you mean to use this variable in the next line instead of `indexingDateString`?

Reply:
Oh yikes - when I tested this it appeared to work and this would definitely not work, so I'll need to take another look at this, I think.

Reply:
Okay, after some quick testing, I've now fixed this and confirmed that it works.